x-pack/plugins/infra/public/pages/logs/log_entry_rate/sections/anomalies/log_entry_example.tsx

```diff
@@ -0,0 +1,279 @@
+  const encodedDatasetFieldValue = useMemo(
+    () => JSON.stringify(getFriendlyNameForPartitionId(dataset)),
+    [dataset]
+  );
+
+  const viewInStreamLinkProps = useLinkProps({
+    app: 'logs',
+    pathname: 'stream',
+    search: {
+      logPosition: encode({
+        end: moment(timeRange.endTime).format('YYYY-MM-DDTHH:mm:ss.SSSZ'),
+        position: { tiebreaker, time: timestamp },
+        start: moment(timeRange.startTime).format('YYYY-MM-DDTHH:mm:ss.SSSZ'),
+        streamLive: false,
+      }),
+      flyoutOptions: encode({
+        surroundingLogsId: id,
+      }),
+      logFilter: encode({
+        expression: `${partitionField}: ${dataset}`,
+        kind: 'kuery',
+      }),
+    },
+  });
+
+  const viewAnomalyInMachineLearningLinkProps = useLinkProps(
+    getEntitySpecificSingleMetricViewerLink(jobId, timeRange, {
+      [partitionField]: dataset,
+    })
+  );
+
+  const menuItems = useMemo(() => {
+    if (!viewInStreamLinkProps.onClick || !viewAnomalyInMachineLearningLinkProps.onClick) {
+      return undefined;
+    }
+
+    return [
+      {
+        label: VIEW_IN_STREAM_LABEL,
+        onClick: viewInStreamLinkProps.onClick,
+      },
+      {
+        label: VIEW_ANOMALY_IN_ML_LABEL,
+        onClick: viewAnomalyInMachineLearningLinkProps.onClick,
+      },
+    ];
+  }, [viewInStreamLinkProps.onClick, viewAnomalyInMachineLearningLinkProps.onClick]);
+
+  return (
+    <LogEntryRowWrapper
+      scale={exampleMessageScale}
+      onMouseEnter={setItemIsHovered}
+      onMouseLeave={setItemIsNotHovered}
+    >
+      <LogEntryColumn {...columnWidths[timestampColumnId]}>
+        <LogEntryTimestampColumn format={exampleTimestampFormat} time={timestamp} />
+      </LogEntryColumn>
+      <LogEntryColumn {...columnWidths[messageColumnId]}>
+        <LogEntryMessageColumn
+          columnValue={{
+            columnId: messageColumnId,
+            message: [{ field: 'message', value: message, highlights: [] }],
+          }}
+          highlights={noHighlights}
+          isActiveHighlight={false}
+          wrapMode="none"
+        />
+      </LogEntryColumn>
+      <LogEntryColumn {...columnWidths[datasetColumnId]}>
+        <LogEntryFieldColumn
+          columnValue={{
+            columnId: datasetColumnId,
+            field: 'event.dataset',
+            value: encodedDatasetFieldValue,
+            highlights: [],
+          }}
+          highlights={noHighlights}
+          isActiveHighlight={false}
+          wrapMode="none"
+        />
+      </LogEntryColumn>
+      <LogEntryColumn
+        key="logColumn iconLogColumn iconLogColumn:details"
+        {...columnWidths[iconColumnId]}
+      >
+        {(isHovered || isMenuOpen) && menuItems ? (
+          <LogEntryContextMenu
+            aria-label={MENU_LABEL}
+            isOpen={isMenuOpen}
+            onOpen={openMenu}
+            onClose={closeMenu}
+            items={menuItems}
+          />
+        ) : null}
+      </LogEntryColumn>
```

Comment:
Maybe? In the design the example log columns weren't perfectly aligned with the row columns, so I (maybe incorrectly) assumed it was okay for those to sit as they are.

Reply:
If it's as easy as I expect, I think it would look cleaner and match the log stream. The column size calculation already gives us the dimensions, after all.

Reply:
👍